src/systems/filecoin_blockchain/struct/block/block.id

```diff
@@ -41,7 +41,7 @@ type BlockHeader struct {
 
 type Block struct {
     Header    BlockHeader
-    Messages  [msg.Message]
+    Messages  [msg.SignedMessage]
```

Comment:
Does this match with what we expect on-chain? My impression was that the Block struct was the motivation for the union -- i.e., some of the messages are BLS signed and appear as UnsignedMessages in the list (with the signature aggregated), while some are Secp signed and appear as SignedMessages.

Reply:
The `Block` type does not go on chain at all. This is an internal implementation detail that's in the spec only because we're trying to write implementation code in it.

But ok, I've added `SECPMessages` and `BLSMessages` separately, which is closer to what an implementation might actually use.

Reply:
This particular definition of the `Block` struct may not go on-chain, but some version of it must. In particular, the `MessageRoot` and `ReceiptRoot` fields in `BlockHeader` are currently uninterpreted CIDs, which really should be modified to refer to the actual underlying algebraic datatype (respectively, `[Message]` and `[MessageReceipt]`). Then the question becomes what to do with that `[Message]`.

In principle we could replace it with two separate arrays as you've done with `SECPMessages` and `BLSMessages`, but then we would lose the ordering of messages within the block (i.e., either all BLS messages must hit the VM before all Secp messages, or vice versa). This seems potentially problematic, and the on-chain union type a cleaner solution.

Reply:
yes, they must be replaced with two different arrays, the spec already had that at one point and both implementations do it.

Reply:
Edit: the spec already has that in the *other* BlockHeader definition (tho it's also wrong for different reasons). I intend to clean those up as part of block validation, for which this is pre-work.

Reply:
OK, sounds good.